LibOS/shim/src/bookkeep/shim_vma.c

```diff
@@ -412,6 +412,17 @@ __set_vma_comment (struct shim_vma * vma, const char * comment)
     vma->comment[len] = 0;
 }
 
+/*
+ * Add bookkeeping for mmap(). "prev" must be passed into the function as
+ * the immediate precedent vma of the allocating address, or NULL if no
+ * vma is lower than the address. If the bookkeeping area overlaps with
```

Comment:
Hmm, this doesn't match with the usage in `init_vma`:
https://github.com/oscarlab/graphene/blob/eddc7b7ce512ecdd9768f684f90dcd4e5c1eab08/LibOS/shim/src/bookkeep/shim_vma.c#L269-L279
The code calls this function with `prev == NULL` without checking if there is any lower vma already allocated.

Reply:
True. This is a violation of the semantic. I have created a wrapper function for adding bookkeeping at the early stage.